test/unit/ngsiv2/examples/contextAvailabilityRequests/registerIoTAgent1.json

```diff
@@ -15,5 +15,6 @@
             "url": "http://smartGondor.com"
         },
         "legacyForwarding": true
-    }
+    },
+    "expires":"2015-09-05T07:35:01.468Z"
```

Comment:
Expiration in the past? Is this intentional (maybe for a test that check something related with expirations...)?

Reply:
I have seen the same date in 2015 in many others...

Reply:
I will try to explain the reason behind this:
- In NGSIv1, the registration payload included a field called `duration` that was typically copied from the configuration file as you can see in https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/services/devices/registrationUtils.js#157.
- In NGSIv2, this field is replaced by `expires` so if it any more time duration but a fixed date. You can see this in https://github.com/telefonicaid/iotagent-node-lib/blob/master/lib/services/devices/registrationUtils.js#341.
- The problem when running unit tests is that the `expires` field is not fixed but depends on the moment when the tests are executed. We could include a complex validation function to check this field but the original developers of the IOTAs used for this [timekeeper](https://www.npmjs.com/package/timekeeper): *This module mocks Date and Date.now in order to help you test time-dependent code. .*
-You can see in https://github.com/telefonicaid/iotagent-node-lib/blob/master/test/unit/ngsiService/active-devices-test.js how they used this module. For instance in https://github.com/telefonicaid/iotagent-node-lib/blob/master/test/unit/ngsiService/active-devices-test.js#203.
- So we could force the date to any time, I have just used this date in 2015 to be consistent with the original NGSIv1 unit tests.

Reply:
Ok. I didn't remember about the mocked date in the past. It makes sense. Thanks for the clarification!

NTC